tests/source/non-lifetime-binders.rs

```diff
@@ -0,0 +1 @@
+fn main() where for<'a, T: Sized + 'a, const C: usize> [&'a T; C]: Sized { }
```

Comment:
I don't know what is conventional for tests.

I can also add the other syntax positions where these can show up, i.e. in `dyn for<T> ..` and `for<T> fn(..)`.

Reply:
I haven't had a chance to review this, but I would definitely say that adding additional cases where these show up would be beneficial!

Reply:
If you get a chance can you add a test case for the code snippet in #5721. I believe this PR should resolve that issue.

Reply:
That code snippet should be covered by the test I added. Do you want me to add another test to match that one exactly?

Reply:
I think adding a `issue_5721.rs` test would be useful

Reply:
Done.